chain/chain/src/flat_storage_resharder.rs

```diff
@@ -320,6 +335,14 @@ impl FlatStorageResharder {
             }
         };
 
+        #[cfg(feature = "test_features")]
+        {
+            if self.adv_should_delay_task(&resharding_hash, chain_store) {
+                info!(target: "resharding", "flat storage shard split task has been artificially postponed!");
```

Comment:
I wonder if it would make sense in the future to try and implement a more general version of this in the test loop API, where you could ask the test loop to not poll some future until a little while later. Idk how it would look exactly, but maybe you could give the testloop user code a way to inspect the future that's about to be polled and decide to move it back in the queue or something, idk

Reply:
Oh that would be very nice. Maybe not applicable in all cases but a strong tool indeed.